### PR TITLE
[WIP] Enable rootless integration tests

### PR DESCRIPTION
**Pull request by baude**
Signed-off-by: baude <bbaude@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
If this works. Awesome.
LGTM

**Comment by baude** [comment]
its going to take some doing @rhatdan but thats the idea

**Comment by TomSweeneyRedHat** [comment]
LGTM and I'm keeping all the fingers and toes crossed for happy tests.

File: contrib/cirrus/rootless_test.sh [hunk 1]
```diff
@@ -29,6 +29,8 @@ case "${OS_RELEASE_ID}-${OS_RELEASE_VER}" in
     fedora-29) ;&
     fedora-28)
         make
+        make install
+        make localintegration
         ;;
     *) bad_os_id_ver ;;
 esac
```
Comment (cevich) [inline contrib/cirrus/rootless_test.sh]: I say just kill this whole case statement.  Make life simpler :smile:

File: contrib/cirrus/setup_environment.sh [hunk 2]
```diff
@@ -79,6 +79,10 @@ then
     if run_rootless
     then
         setup_rootless
+        make install.catatonit
+        go get github.com/onsi/ginkgo/ginkgo
+        go get github.com/onsi/gomega/...
+        dnf -y update runc
```
Comment (cevich) [inline contrib/cirrus/setup_environment.sh]: These things should probably all be done while building VM images.  No reason to eat that extra time for every PR.  I can help sort that out later though.

**Comment by cevich** [comment]
@baude in case you missed it on IRC:

* Matt said the cgroups tests are bugged (they should report an error, but instead silently ignore).  So skip for now, w/ comment and/or open issue.
* The `nsenter` tests should work, Matt suspects a real bug, I suspect a test-bug :smile: 
* The networking tests are going to be hit/miss, we'll have to knock 'em down as they come.

**Comment by cevich** [comment]
@baude FYI- I force-pushed to #2694 with a few small updates.  With that PR and this one, you should now be able to: ``hack/get_ci_vm.sh -r fedora-29-libpod-6318419153518592`` and it'll plop you right into the rootless user (after running the setup).

**Comment by baude** [comment]
@giuseppe is running with this in https://github.com/containers/libpod/pull/2704 ... closing this out and will merge his with the rootless fixes.

**Pull request closed without merging**